Mathlib/Logic/Function/Basic.lean

```diff
@@ -141,6 +141,12 @@ theorem Injective.of_comp_iff' (f : α → β) {g : γ → α} (hg : Bijective g
   λ h => h.comp hg.injective⟩
 #align function.injective.of_comp_iff' Function.Injective.of_comp_iff'
 
+theorem Injective.of_comp_right {g : γ → α} (I : Injective (f ∘ g)) (hg : Surjective g) :
+    Injective f := fun x y h ↦ by
+  obtain ⟨x, rfl⟩ := hg x
+  obtain ⟨y, rfl⟩ := hg y
+  exact congr_arg g (I h)
```

Comment:
I think you can use this to golf the theorem above if you swap the order.

Reply:
Thanks! Golfed the surjective counterpart as well.